rs/crypto/internal/crypto_lib/tls/src/lib.rs

```diff
@@ -148,12 +143,12 @@ fn x509_v3_certificate(
     cert_result
 }
 
-fn rcgen_keypair_from_ed25519_keypair(
-    secret_key: &ed25519_types::SecretKeyBytes,
-    public_key: &ed25519_types::PublicKeyBytes,
+fn rcgen_keypair_from_ed25519_key(
+    secret_key: &PrivateKey,
 ) -> Result<KeyPair, TlsKeyPairAndCertGenerationError> {
-    let keypair_der = secret_key_to_pkcs8_v2_der(secret_key, public_key);
-    KeyPair::try_from(keypair_der.expose_secret()).map_err(|e| {
+    let keypair_der = secret_key.serialize_pkcs8(PrivateKeyFormat::Pkcs8v2);
```

Comment:
The `serialize_pkcs8` function does support encoding in the previously used format via `PrivateKeyFormat::Pkcs8v2WithRingBug`. However the version of ring we are now using is recent enough to also support the correct encoding, so I went with that.